### PR TITLE
Rename a few attributes

### PR DESCRIPTION
**Pull request by AurelienGasser**
Companion PRs:

- https://github.com/SubstraFoundation/substra-backend/pull/345
- https://github.com/SubstraFoundation/substra-chaincode/pull/129
- https://github.com/SubstraFoundation/substra/pull/241

Rename:

| Before | After |
|---|----|
| `compute_plan_id` | `compute_plan_key` (or simply `key`) |
|  `hash` | `checksum` |
| `data_manager.keys` | `data_manager.data_sample_keys` |

### CODE REVIEW

**Review by jmorel — APPROVED** [review]

File: substratest/client.py [hunk 1]
```diff
@@ -209,10 +209,7 @@ def wait(self, asset, timeout=cfg.FUTURE_TIMEOUT, raises=True):
             assert False, 'Future not supported'
         getter = getattr(self, m)
 
-        if asset.__class__.__name__ == "ComputePlan":
-            key = asset.compute_plan_id
-        else:
-            key = asset.key
+        key = asset.key
```
Comment (jmorel) [inline substratest/client.py:212]: Can we just remove this variable and use asset.key everywhere?

**Review by jmorel — CHANGES_REQUESTED** [review]

File: substratest/client.py [hunk 2]
```diff
@@ -80,8 +80,8 @@ def add_compute_plan(self, spec, *args, **kwargs):
     def update_compute_plan(self, spec, *args, **kwargs):
         spec_dict = spec.dict()
         # Remove extra field from data
-        spec_dict.pop("compute_plan_id")
-        return self._client.update_compute_plan(spec.compute_plan_id, spec_dict, *args, **kwargs)
+        spec_dict.pop("key")
+        return self._client.update_compute_plan(spec.key, spec_dict, *args, **kwargs)
```
Comment (Kelvin-M) [inline substratest/client.py:84]: Why did you rename it as `key` and not `compute_plan_key` ? 
This modification occurs multiple time in this PR :)

**Review by Kelvin-M — CHANGES_REQUESTED** [review]

File: substratest/client.py (hunk 2 above)
Comment (AurelienGasser) [reply substratest/client.py:84]: Comment addressed in https://github.com/SubstraFoundation/substra/pull/241#issuecomment-721759361

File: substratest/client.py (hunk 1 above)
Comment (AurelienGasser) [reply substratest/client.py:212]: The only place where we use this variable is 

```
asset = getter(key)
```

I think changing it to 

```
asset = getter(asset.key)
```

might potentially bring new categories of bugs: if `getter` doesn't return the correct result, then the subsequent `getter` call will use the wrong argument. For this reason I think it's better to create a dedicated variable `key` to ensure `key` is constant throughout the execution of this function.

What do you think?

Comment (jmorel) [reply substratest/client.py:212]: Makes sense, thank you for the explanation!

**Review by jmorel — APPROVED** [review]

**Comment by AurelienGasser** [comment]
The travis-ci tests won't pass until the substra PR is merged.

**Review by Kelvin-M — APPROVED** [review]
